### DISCUSSION TITLE
No ABL Language Server

### DISCUSSION THREAD

**Discussion by CharltonAwesome1** [vscode-abl/vscode-abl#51] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/80033031/212048100-3af6bc1d-fe70-4d11-8c1c-aaf468735f15.png)

Can someone please help me with this? I'm not sure how to fix it.

Things I've tried:
* Using the docs on code.visualstudio.com related to language servers. There was no info on installing one
* Downloading the git repo https://github.com/vscode-abl/vscode-abl and running npm install on it
* Restarting Language Server in VSCode using all the options

Things I've done:
* Installed the extension from extensions in VSCode
* Installed and configured the JVM line of code, as well as setting up abl.langServerJavaExecutable
* Updated the debugger files

I'm basically just trying to use the Intellisense function for ABL in VSCode, as well as being able to run/debug programs.

**Comment by gquerret** [comment, marked as answer]
Did you open VSCode in a directory where you have a project configuration file ? Check `README.md` on the main page.

**Reply by CharltonAwesome1** [reply]
I ran the [code](https://github.com/cverbiest/pct-utils/blob/master/src/vscode/generate_riverside_vscodeconfig.p) successfully.

I'm sorry. I'm not sure what you mean by "Did you open VSCode in a directory where you have a project configuration file ?".

I swear I'm not dumb. Just confused.

**Reply by gquerret** [reply]
> I'm sorry. I'm not sure what you mean by "Did you open VSCode in a directory where you have a project configuration file ?".

If you have generated your openedge-project.json file, make sure it is in the root directory of this project. Then start VSCode from the command-line using `code .`. Or open VSCode, and use File -> Open Folder, and select the root directory of the project. That will start the language server.

> I swear I'm not dumb. Just confused.

🤣

**Reply by CharltonAwesome1** [reply]
I pasted the openedge-project.json in the same directory as the openedge (.p) file I'm working on. That fixed everything.
Thank you.
![image](https://user-images.githubusercontent.com/80033031/212132294-24f6dc03-a8f8-4850-ac0a-84b40b45e410.png)

I still have two questions.
* How do I connect a database. (I know that it I need to add something to "dbConnections" in the json file. I'm just not sure of the format and how it should point to the folder in which the database is)
* How do I fix this:
![image](https://user-images.githubusercontent.com/80033031/212132526-d3e727d6-26cb-4a67-a57d-57aabfd6128c.png)
Is it a simple port number change in the debugger file?

I'm not sure if these deserve their own threads. And I'm don't know if they sound silly. This is my first time using Openedge ABL.

**Reply by CharltonAwesome1** [reply]
> Did you open VSCode in a directory where you have a project configuration file ? Check `README.md` on the main page.

This should be question with the "answered" tag . Thanks.

**Comment by gquerret** [comment]
It would be a bit off-topic to answer DB connection questions here. I know the VSCode plugin documentation has some assumptions about OpenEdge in general, but you can ask generic OpenEdge questions on [Progress Community](https://community.progress.com).  In short, it's probably something like `-db dbName -H hostName -S portNumber`, but then there are lots of variations.

The default port number for the debugger is 9999. The [documentation](https://github.com/vscode-abl/vscode-abl/blob/8251cde5665acf9121b11663ce5952d8baa92e9e/README.md#debugger) says that it's hard-coded, but that will change in the future.

**Reply by CharltonAwesome1** [reply]
Great Answer. Thank you very much.
